app/jobs/reports/base_report.rb

```diff
@@ -47,13 +47,28 @@ def transaction_with_timeout(rails_env = Rails.env)
     end
 
     def save_report(report_name, body, extension:)
+      write_json_report_to_analytics(report_name, body, extension)
       if !IdentityConfig.store.s3_reports_enabled
         logger.info('Not uploading report to S3, s3_reports_enabled is false')
         return body
       end
       upload_file_to_s3_timestamped_and_latest(report_name, body, extension)
     end
 
+    def write_json_report_to_analytics(report_name, body, extension)
+      return unless extension == 'json'
+      build_analytics.track_event(
+        Analytics::REPORT_RESULTS,
+        user_id: '',
+        report_name: report_name,
+        report_body: JSON.parse(body),
```

Comment:
Is this writing the entire report body to `events.log`?

is that generally a safe thing to do? and even if it's there, are the report bodies in a form that we can query accurately with Cloudwatch Insights queries? A lot of our reports have arrays which are notoriously difficult to query in Cloudwatch Insights... do you have any example queries?

Reply:
I don't think we should log all report bodies to Cloudwatch. If there are specific reports that need to be more accessible, is it possible to put a simple page up on data.login.gov?

Reply:
There's never any PII in the reports so it's fine.  Cloudwatch queries can parse JSON.  I thought about this a bit and did consider writing it only for the user counts which is all I needed.  However the value of having every report simultaneously as a time based query in cloudwatch is very valuable esp. without the need to write any additional idp code now in and the future.  I'm weighing this vs the complexity of the cloudwatch queries.  I'm going to punch some reports into the logs to int and play around with it first.

Reply:
@mitchellhenke can you be specific as to why we should not log them to analytics?  A majority of the reports run once daily and there is no PII.

Reply:
Yep, thanks for adding some context for me!

I think since we have these reports in S3, putting them in CloudWatch is duplicative and I think (?) that all events we log right now are generated by an individual user taking some action in the IDP. I'd prefer we keep `events.log` as the destination for those kinds of logs. Being able to query results across time is a benefit, but I think it may be kind of limited since the time ranges that are query-able in one shot without timing out is on the order of days or weeks with CloudWatch having to scan through all of the other events that get logged.

For me, I see two other options that may be more preferable. If we're primarily after easier access to user counts, my preference would be to put this data on data.login.gov if possible. The other less appealing and more involved option would be to have a new `reports.log` file in CloudWatch where we put these.

Reply:
It is a given that we will be using cloudwatch to visualize the counts per sp and globally. Wrt to going to a new log file it could be argued that there are other things we are logging in idp that could go to specialized log files for the sake of efficiency of queries ie VENDOR_OUTAGE. Moreover this workload is read-few and we only need to scan 24 hours. However long term I like the idea of a new log file because eventually we will wish to do more than straight counts and show historical counts.

Reply:
In summary: I like new `reports.log` file in CloudWatch.  Adding it now... Excellent idea @mitchellhenke

Reply:
Still dabbling with the JSON queries in cloudwatch.  They appear to be quite robust

Reply:
Oh ok, apologies! I misunderstood and assumed that time-based queries meant the intention was to query daily reports over multiple days, rather than just for a given day. Please do correct me if I'm wrong, but my understanding is vendor outage logging happens when there is an outage, and a person is blocked from continuing due to that outage.

I'm more than happy to help with the separate log file if that's the best path forward 🙂

Reply:
Not sure if vendor outage is a good example but the other candidates would be things we piggy back on the normal user events logging that are actually sporadic in nature (ie the first unique event state: verb/path/event I added to get accurate counts in cloudwatch).  Essentially we are trying to add structure the (denormalized) log data by creating separate log files and the epitome of that would be using cloudwatch metrics or adding all the the events to customized tables in redshift.  The reports log is definitely an ideal candidate for a separate log file.